csharp/src/Ice/ObjectAdapter.cs

```diff
@@ -798,18 +800,34 @@ public void SetPublishedEndpoints(IEnumerable<Endpoint> newEndpoints, Cancellati
                             }
                         }
 
+                        if (endpoints.Any(endpoint => endpoint is IPEndpoint ipEndpoint && ipEndpoint.Port == 0))
+                        {
+                            if (endpoints.Count > 1)
+                            {
+                                throw new InvalidConfigurationException(@$"object adapter `{Name
+                                    }': only one endpoint is allowed when a dynamic IP port (:0) is configured");
+                            }
+                            if (endpoints[0] is IPEndpoint ipEndpoint && ipEndpoint.Address == IPAddress.None)
+                            {
+                                throw new InvalidConfigurationException(@$"object adapter `{Name
+                                    }': use an IP address to configure an endpoint with a dynamic port (:0)");
```

Comment:
Should we detect this earlier when we parse ObjectAdapter endpoints?

Reply:
I think cleaner to keep the parsing code fairly dumb and do this check right after parsing and not during parsing.